From d16600352c56d08cff94d95774791a6556d0d6ba Mon Sep 17 00:00:00 2001
From: Luca Olivetti <luca@wetron.es>
Date: Fri, 5 Feb 2016 12:02:51 +0100
Subject: [PATCH 1072/1096] pdb: Fix segfault in pdb_ldap for missing gecos

Bug: https://bugzilla.samba.org/show_bug.cgi?id=11530
Signed-off-by: Luca Olivetti <luca@wetron.es>
Reviewed-by: Volker Lendecke <vl@samba.org>
Reviewed-by: Alexander Bokovoy <ab@samba.org>

Autobuild-User(master): Volker Lendecke <vl@samba.org>
Autobuild-Date(master): Fri Feb  5 16:47:00 CET 2016 on sn-devel-144

(cherry picked from commit 5d759bd0d4bf7cae8b54b69af5ecacb7987c2a0f)

Autobuild-User(v4-4-test): Karolin Seeger <kseeger@samba.org>
Autobuild-Date(v4-4-test): Thu May 12 11:48:24 CEST 2016 on sn-devel-144

(cherry picked from commit 9f864190b0eae385cd02b45aa632bf9321aaaea0)
---
 source3/passdb/pdb_ldap.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/source3/passdb/pdb_ldap.c b/source3/passdb/pdb_ldap.c
index c65fb08..50ab3a7 100644
--- a/source3/passdb/pdb_ldap.c
+++ b/source3/passdb/pdb_ldap.c
@@ -1005,7 +1005,7 @@ static bool init_sam_from_ldap(struct ldapsam_privates *ldap_state,
 				entry,
 				"gecos",
 				ctx);
-		if (unix_pw.pw_gecos) {
+		if (unix_pw.pw_gecos == NULL) {
 			unix_pw.pw_gecos = fullname;
 		}
 		unix_pw.pw_dir = smbldap_talloc_single_attribute(
@@ -1013,7 +1013,7 @@ static bool init_sam_from_ldap(struct ldapsam_privates *ldap_state,
 				entry,
 				"homeDirectory",
 				ctx);
-		if (unix_pw.pw_dir) {
+		if (unix_pw.pw_dir == NULL) {
 			unix_pw.pw_dir = discard_const_p(char, "");
 		}
 		unix_pw.pw_shell = smbldap_talloc_single_attribute(
@@ -1021,7 +1021,7 @@ static bool init_sam_from_ldap(struct ldapsam_privates *ldap_state,
 				entry,
 				"loginShell",
 				ctx);
-		if (unix_pw.pw_shell) {
+		if (unix_pw.pw_shell == NULL) {
 			unix_pw.pw_shell = discard_const_p(char, "");
 		}
 
-- 
2.9.0

